cmd/metacache-set.go

```diff
@@ -590,25 +590,73 @@ func (er *erasureObjects) streamMetadataParts(ctx context.Context, o listPathOpt
 func getListQuorum(quorum string, driveCount int) int {
 	switch quorum {
 	case "disk":
-		// smallest possible value, generally meant for testing.
 		return 1
 	case "reduced":
 		return 2
 	case "optimal":
 		return (driveCount + 1) / 2
+	case "auto":
+		return -1
 	}
 	// defaults to 'strict'
 	return driveCount
 }
 
+func getQuorumDisks(disks []StorageAPI, infos []DiskInfo, readQuorum int) (newDisks []StorageAPI) {
```

Comment:
Question: "auto" list quorum will work only if we see some common mutations in the disks? ie, if the disks go down and come up randomly, we won't be able to get some common mutations on (total deletes + total writes), and we end up choosing all the disks for the listing. Correct?

Reply:
Idea is that eventually all the missed events will get healed. Very flaky systems we can't do much they will behave the same as what it is today.

Remember this is erasure set level what it means is that there are many other erasure sets that may be good so overall it shall benefit. If one or two sets don't qualify it isn't a big deal

Reply:
Also for now this feature is kept opt-in, default will be still strict.